qemu/unicorn_common.h

```diff
@@ -55,9 +56,48 @@ static void release_common(void *t)
     memory_free(uc);
 
     // Clean CPU.
+    CPU_FOREACH(cpu) {
+        g_free(cpu->tcg_as_listener);
+        g_free(cpu->thread);
+        g_free(cpu->halt_cond);
+    }
+
+    OBJECT(uc->machine_state->accelerator)->ref = 1;
+    OBJECT(uc->machine_state)->ref = 1;
+    OBJECT(uc->owner)->ref = 1;
+    OBJECT(uc->root)->ref = 1;
+
+    object_unref(uc, OBJECT(uc->machine_state->accelerator));
+    object_unref(uc, OBJECT(uc->machine_state));
     object_unref(uc, uc->cpu);
+    object_unref(uc, OBJECT(&uc->io_mem_notdirty));
+    object_unref(uc, OBJECT(&uc->io_mem_unassigned));
+    object_unref(uc, OBJECT(&uc->io_mem_rom));
+    object_unref(uc, OBJECT(uc->root));
     g_hash_table_foreach(uc->type_table, free_table, uc);
 
+    g_free(uc->system_memory);
+
+    if(uc->qemu_thread_data)
+        free(uc->qemu_thread_data);
+
+#if TCG_TARGET_REG_BITS == 32
+    for(int i = 0; i < s->nb_globals; i++)
+    {
+        TCGTemp *ts = &s->temps[i];
+        if(ts->base_type == TCG_TYPE_I64)
+        {
+            if(ts->name && ((strcmp(ts->name+(strlen(ts->name)-2), "_0") == 0) || (strcmp(ts->name+(strlen(ts->name)-2), "_1") == 0)))
```

Comment:
These global temps are created in tcg_global_mem_new_internal and are suffixed with _0 and _1 respectively, when the host is 32-bits and a 64-bit type is needed.